### DISCUSSION TITLE
Enabling cbc ciphers

### DISCUSSION THREAD

**Discussion by FernS3** [drakkan/sftpgo#646] [Q&A] [answered]
Hi Nicola, 

I am running this SFTP server and trying to connect from a legacy client.

The legacy client is attempting to use aes256-cbc cipher when connecting , which is now not active by default.

On the host where I have docker running I have that specific cipher enabled in /etc/ssh/ssh_config but this does not seem to be working in the docker container itself. 

Are there currently any ways to enable additional ciphers to the docker container?

**Comment by drakkan** [comment, marked as answer]
Hi,

you can customize [the ciphers](https://github.com/drakkan/sftpgo/blob/main/sftpgo.json#L65). An empty list enables the default ciphers:

- aes128-gcm@openssh.com
- aes256-gcm@openssh.com
- chacha20-poly1305@openssh.com
- aes128-ctr
- aes192-ctr 
- aes256-ctr

you can also set `aes128-cbc`, `3des-cbc` and arc four based ciphers.

Please note that CBC mode is insecure and so is not included in the default config (see https://www.ieee-security.org/TC/SP2013/papers/4977a526.pdf), you should expect that an active attacker can recover plaintext if you use it

**Reply by FernS3** [reply]
Thank you for your response! 

I have let the folks know this is a security risk but it seems to be a risk we are willing to take.
